src/connectionController.ts

```diff
@@ -441,7 +441,7 @@ export default class ConnectionController {
         this.eventEmitter.emit(DataServiceEventTypes.CONNECTIONS_DID_CHANGE);
         this.eventEmitter.emit(DataServiceEventTypes.ACTIVE_CONNECTION_CHANGED);
 
-        if (this._telemetryController) {
+        if (this._telemetryController?.needTelemetry()) {
```

Comment:
And here  - I think because we check if (this.needTelemetry()) { in the telemetry controller we can remove this if?

Reply:
This one we actually want to keep, since `this.sendTelemetry()` requests cloud info what takes a while and can cause a timeout in tests. But I removed the rest of these checks from other controllers.